src/picologging/handlers.py

```diff
@@ -464,3 +464,57 @@ def doRollover(self):
                     addend = 3600
                 now_rollover_at += addend
         self.rollover_at = now_rollover_at
+
+
+class BufferingHandler(picologging.Handler):
+    """
+    A handler class which buffers logging records in memory. Whenever each
+    record is added to the buffer, a check is made to see if the buffer should
+    be flushed. If it should, then flush() is expected to do what's needed.
+    """
+    def __init__(self, capacity):
+        """
+        Initialize the handler with the buffer size.
+        """
+        picologging.Handler.__init__(self)
+        self.capacity = capacity
+        self.buffer = []
+
+    def shouldFlush(self, record):
+        """
+        Should the handler flush its buffer?
+        Returns true if the buffer is up to capacity. This method can be
+        overridden to implement custom flushing strategies.
+        """
+        return (len(self.buffer) >= self.capacity)
+
+    def emit(self, record):
+        """
+        Emit a record.
+        Append the record. If shouldFlush() tells us to, call flush() to process
+        the buffer.
+        """
+        self.buffer.append(record)
+        if self.shouldFlush(record):
```

Comment:
Inline `.shouldFlush()` into this function and remove that method altogether. The overhead of the extra method call isn't worth the abstraction

Reply:
There's one drawback that subclassing `BufferingHandler` you should override `emit` to customize the flushing logic altogether.
If that's acceptable then I'll just update the PR.

Reply:
The performance gain outweighs the possible customization IMO.

I don't feel like `shouldFlush()` needs to be part of the API (even though it is a public method).

Reply:
Done.